vunit/builtins.py

```diff
@@ -68,13 +68,55 @@ def _add_files(self, pattern):
 
             self._vunit_lib.add_source_file(file_name)
 
-    def _add_data_types(self):
+    def _add_data_types(self, use_external=None, impls=None):
```

Comment:
The interface seems wierd, why not just have a list of implementation where all of them are enabled by being mentioned in the list itself. Disabling is the absence from the list. Why have a separate boolean array?

Reply:
We have three possible use cases:

- `*vhpi.vhd` and `*body.vhd`.
- `*novhpi.vhd` and `*body.vhd`.
- `custom.vhd` and `custom-body.vhd`.

`*novhpi.vhd`, `*vhpi.vhd` and `*body.vhd` are part of the VUnit codebase; `custom.vhd` and `custom-body.vhd` are not.

On most cases the user will want to select whether to enable external models or not, and he/she will not want to know where are the sources located in the Vunit codebase. Only in marginal cases will he/she want to provide an alternative implementation.

However, your answer made me realize that python is a dynamic language. Hence, I can use a single list to hold either `None`, `Boolean` or a list of strings.

I will implement this.